scripts/algorithms/fm.dml

```diff
@@ -0,0 +1,83 @@
+#-------------------------------------------------------------
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+# 
+#   http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+# Implements the factorization machines (2-way)
+#
+#
+#
+# ------------------------------------------------------------------------------------------------------
+# NAME      TYPE     DEFAULT  MEANING
+# ------------------------------------------------------------------------------------------------------
+# X         String   ---      Location to read the input matrix X to be factorized
+# y         String   ---      Location to read the input matrix y corresponding to the feature vector
+# k         Int      ---      hyperparameter
+# reg       String   "L2"	    Regularization: 
+#                             "L2" = L2 regularization;
+#                             "wL2" = weighted L2 regularization
+# lambda_w  Double   0.000001 Regularization parameter, no regularization if 0.0
+# lambda_v  Double   0.000001
+# y_est     String   ---      Output file for estimated target values
+# error 
+# -------------------------------------------------------------------------------------------------------
+# Example: An extended design matrix with four real valued predictors. 
```

Comment:
@dusenberrymw can you find a small example for me on FM's?.    
https://gist.github.com/p-janardhan/a05f022d276090637c59f541d94beccf